node/impl/full/eth.go

```diff
@@ -699,21 +701,29 @@ func (a *EthModule) EthFeeHistory(ctx context.Context, p jsonrpc.RawParams) (eth
 	)
 
 	for blocksIncluded < int(params.BlkCount) && ts.Height() > 0 {
-		msgs, rcpts, err := messagesAndReceipts(ctx, ts, a.Chain, a.StateAPI)
-		if err != nil {
-			return ethtypes.EthFeeHistory{}, xerrors.Errorf("failed to retrieve messages and receipts for height %d: %w", ts.Height(), err)
-		}
+		basefee = ts.Blocks()[0].ParentBaseFee
```

Comment:
This is a bug fix. We need to include this regardless of whether or not we merge this patch.